From a78bd36e3973b971664b59b77711be773155d342 Mon Sep 17 00:00:00 2001
From: Balamurugan Ramar <quic_bramar@quicinc.com>
Date: Thu, 10 Apr 2025 23:26:47 +0530
Subject: [PATCH] mac80211: compilation fixes for backports .

1.Added appropriate ENUM for chan width,
2.Assigned monitor_chandef[hw_idx] to dflt_chandef to fix compilation
3.Modified wiphy_work API's to work_struct in API's to fix compilation failures

Signed-off-by: Kaviarasu Sanjeevi <quic_kaviaras@quicinc.com>
Signed-off-by: Balamurugan Ramar <quic_bramar@quicinc.com>
---
 net/mac80211/main.c | 5 ++---
 net/mac80211/util.c | 3 +++
 2 files changed, 5 insertions(+), 3 deletions(-)

--- a/net/mac80211/main.c
+++ b/net/mac80211/main.c
@@ -490,11 +490,10 @@ static void ieee80211_restart_work(struc
 	/* we might do interface manipulations, so need both */
 	wiphy_lock(local->hw.wiphy);
 	wiphy_work_flush(local->hw.wiphy, NULL);
-
-	wiphy_work_flush(local->hw.wiphy, &local->sched_scan_stopped_work);
-        wiphy_work_flush(local->hw.wiphy, &local->radar_detected_work);
         flush_work(&local->awgn_detected_work);
 
+	wiphy_work_cancel(local->hw.wiphy, &local->sched_scan_stopped_work);
+	wiphy_work_cancel(local->hw.wiphy, &local->radar_detected_work);
 	WARN(test_bit(SCAN_HW_SCANNING, &local->scanning),
 	     "%s called with hardware scan in progress\n", __func__);
 
--- a/net/mac80211/util.c
+++ b/net/mac80211/util.c
@@ -3782,6 +3782,7 @@ static void ieee80211_awgn_detected_proc
 	struct ieee80211_chanctx *ctx;
 	int num_chanctx = 0;
 
+	lockdep_assert_wiphy(local->hw.wiphy);
 	list_for_each_entry(ctx, &local->chanctx_list, list) {
 		if (ctx->replace_state == IEEE80211_CHANCTX_REPLACES_OTHER)
 			continue;
@@ -3906,6 +3907,7 @@ void ieee80211_dfs_radar_detected_work(s
 	u16 radar_bitmap;
 	struct ieee80211_channel *radar_channel = NULL;
 
+	lockdep_assert_wiphy(local->hw.wiphy);
 	if (list_empty(&local->radar_info_list))
 		return ieee80211_dfs_radar_detected_processing(local, 0,
 							       radar_channel);
